src/main/java/de/dennisguse/opentracks/AbstractTrackDeleteActivity.java

```diff
@@ -38,21 +37,46 @@
  *
  * @author Jimmy Shih
  */
-//TODO It is actually not a List; we need a better name for this class
 //TODO Check if this class is still such a good idea; inheritance might limit maintainability
-public abstract class AbstractListActivity extends AbstractActivity implements ConfirmDeleteCaller {
+public abstract class AbstractTrackDeleteActivity extends AbstractActivity implements ConfirmDeleteCaller {
 
-    private static final String TAG = AbstractListActivity.class.getSimpleName();
+    private static final String TAG = AbstractTrackDeleteActivity.class.getSimpleName();
 
     protected static final int GPS_REQUEST_CODE = 6;
-    private static final int DELETE_REQUEST_CODE = 3;
+
+    private TrackDeleteServiceConnection trackDeleteServiceConnection;
+
+    private ArrayList<Track.Id> trackIds;
+
+    private final Runnable bindChangedStatusCallback = new Runnable() {
+        @Override
+        public void run() {
+            if (trackDeleteServiceConnection == null) {
+                return;
+            }
+            TrackDeleteService service = trackDeleteServiceConnection.getServiceIfBound();
+            if (service == null) {
+                return;
+            }
+
+            service.getDeletingStatusObservable().observe(AbstractTrackDeleteActivity.this, status -> onTrackDeleteStatus(status));
+        }
+    };
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        if (trackDeleteServiceConnection == null) {
+            trackDeleteServiceConnection = new TrackDeleteServiceConnection(bindChangedStatusCallback);
+        }
+        trackDeleteServiceConnection.bind(this);
+    }
 
     @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (requestCode == DELETE_REQUEST_CODE) {
-            onTrackDeleted();
-        } else {
-            super.onActivityResult(requestCode, resultCode, data);
+    protected void onStop() {
+        super.onStop();
+        if (trackDeleteServiceConnection != null) {
+            trackDeleteServiceConnection.unbind(this);
```

Comment:
also `trackDeleteServiceConnection = null`

null check not needed.

Reply:
null check if needed because the connection could be unbind before onStop (when deletion service finishes).
Look at line 120.